src/utils/static.ts

```diff
@@ -14,6 +14,7 @@ export async function serveStatic(
   options: ServeStaticOptions,
 ): Promise<false | undefined | null | BodyInit> {
   if (event.request.method !== "GET" && event.request.method !== "HEAD") {
+    event.response.headers.set("allow", "GET, HEAD");
     if (!options.fallthrough) {
       throw createError({
```

Comment:
I think we should move it inside the `if` condition as we only add header alongside the error.